internal/state/resolver/resolver.go

```diff
@@ -102,7 +106,7 @@ func getTargetPort(svc *v1.Service, svcPort int32) (int32, error) {
 		if port.Port == svcPort {
 			val := port.TargetPort.IntValue()
```

Comment:
the target port can be a string like "myport" rather than "8080". In this case, the actual port number can looked up in the pod spec like here https://github.com/nginxinc/kubernetes-ingress/blob/b69bc90cd6d5603f34744780cc5b4ace038c93fc/internal/k8s/utils.go#L105 

I think we should support it but perhaps we can leave it as a FIXME for now?  

sorry, didn't catch that earlier.

Reply:
Thanks for catching. I overlooked this case. 

I can fix it now by looking up the actual port number in the EndpointSlice ports list.

Reply:
Added logic for string target ports: https://github.com/nginxinc/nginx-kubernetes-gateway/pull/221/commits/ccbebd6b1d7e01c58997933ac14082baa95362e1